caikit/runtime/server_base.py

```diff
@@ -75,6 +77,10 @@ def _find_port(cls, start=8888, end=None, host="127.0.0.1"):
                 # port is open
                 return start
 
+    def _shut_down_model_manager(self):
+        """Shared utility for shutting down the model manager"""
+        ModelManager.get_instance().shut_down()
```

Comment:
this is interesting....

we're shutting down an instance of a runtime server, and in the process shutting down a singleton model manager?

Super fine in the context of a container that is running the runtime entrypoint, but this seems like it could muck with tests where a server that shuts down would kill the singleton poll. I think we have plenty of these hanging around:
```
with runtime_test_server(...) as runtime:
```

How do those not just kill the shared model manager's poll?

Reply:
ug, you're probably right. I really hate that we made everything a singleton! There's no reason for these to be singletons and it just screws with unit tests badly

Reply:
I think the reason that this works is that `shut_down()` only stops the polling and the tests that actually test the polling explicitly create temporary `ModelManager` instances (for exactly this need to isolate them)